source/base/parameter_handler.cc

```diff
@@ -1608,6 +1608,31 @@ bool ParameterHandler::read_input (std::istream &input,
                                    const std::string &filename,
                                    const std::string &last_line)
 {
+  try
+    {
+      parse_input(input, filename, last_line);
+      return true;
+    }
+  catch (const ExcIO &exc)
+    {
+      throw;
+    }
+  // This catch block more or less duplicates the old behavior of this function,
+  // which was to print something to stderr for every parsing error (which are
+  // now exceptions) and then return false.
+  catch (const ExceptionBase &exc)
+    {
+      std::cerr << exc.what() << std::endl;
+      return false;
+    }
```

Comment:
Does this code generate a warning about a missing return value because ``parse_input`` could in theory throw something that is not derived from ``ExceptionBase``? If not, why not? I think it would be safer to have a ``return false;`` at the very bottom of the function.